src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/swift/SwiftAPIClient.java

```diff
@@ -28,8 +35,9 @@
     private String token;
 
     public static final Logger LOGGER = Logger.getLogger(SwiftAPIClient.class);
+    private static final String CONTAINER_URL_PATTERN = "%s/%s?path=%s";
 
-    public SwiftAPIClient(Properties properties) throws SwiftPermanentStorageException {
+    SwiftAPIClient(Properties properties) throws SwiftPermanentStorageException {
```

Comment:
I known you guys are reusing code but ... we should avoid starting threads in the constructor (it is not safe, generally speaking).

It also is usually odd (in a design point-of-view) because then, the constructor is doing more than constructing the object

Reply:
I removed it from constructor and created a method called `run` on Permanent Storage with the function of starting routines and to perform operations that are not suitable in the constructor.